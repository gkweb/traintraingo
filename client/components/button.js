import styled from 'styled-components'

export const ButtonElem = styled.button `
  display: block;
  box-sizing: border-box;
  background-color: ${props => props.theme.secondaryBg};
  border: 1px solid ${props => props.theme.secondaryBg};
  color: ${props => props.theme.primary};

  &:focus {
    outline: 0;
    box-shadow: 0 0 ${props => props.theme.primary} 2px;
  }

  &:hover,
  &:focus {
    background-color: ${props => props.theme.tertiaryBg};
  }
`
