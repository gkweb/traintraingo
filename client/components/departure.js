import styled from 'styled-components'
import moment from 'moment'
import Disruption from './disruption'
import RouteStops from './route-stops'
import GoodService from './good-service'
import PropTypes from 'prop-types'

const ContainerElem = styled.div`
  padding: 1.5em 1em 1em;
  width: 100%;
  border: 1px solid ${props => props.theme.tertiaryBg};
  background: ${props => props.theme.inversePrimaryBg};
  color: ${props => props.theme.inversePrimary};
  box-shadow: -0.0625em 0.125em 0.5em -0.125em rgba(0, 0, 0, 0.125);
`

const TitleElem = styled.h2`
  margin: 0 0 0.75em;
`

const InfoElem = styled.p`
  margin: 0;

  &:not(:last-child) {
    margin: 0 0 1em;
  }
`

const DepartingTimeElem = styled.p`
  font-size: 1.5em;
  margin: 0 0 0.75em;
`

const PlatformElem = styled.p`
  font-size: 1.125em;
  margin: 0;

  &:not(:last-child) {
    margin: 0 0 1em;
  }
`

const Departure = ({
  lineName,
  directionName,
  platformNumber,
  scheduledDep,
  estimatedDep,
  disruptions,
  runId,
}) => {
  const scheduled = moment(scheduledDep).format('HH:mm')
  const estimated = estimatedDep ? moment(estimatedDep).fromNow() : '-' // Estimates into the future are null

  return (
    <ContainerElem>
      <TitleElem>
        {scheduled} to {directionName}
      </TitleElem>
      {lineName ? <InfoElem>Line: {lineName}</InfoElem> : null}
      <DepartingTimeElem>{estimated}</DepartingTimeElem>
      <PlatformElem>
        Platform: {platformNumber ? platformNumber : '-'}
      </PlatformElem>
      <RouteStops runId={runId} />
      {disruptions && disruptions.length > 0 ? (
        <Disruption disruptionData={disruptions} />
      ) : (
        <GoodService />
      )}
    </ContainerElem>
  )
}

Departure.propTypes = {
  lineName: PropTypes.string,
  directionName: PropTypes.string,
  platformNumber: PropTypes.number,
  scheduledDep: PropTypes.string,
  estimatedDep: PropTypes.string,
  disruptions: PropTypes.array,
  runId: PropTypes.number,
}

export default Departure
