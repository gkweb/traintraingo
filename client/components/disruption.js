import React, {Component} from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import {Chevron} from './icon'

const ContainerElem = styled.div`
  display: flex;
  flex-flow: column nowrap;
`

const ChevronIconElem = styled(Chevron)`
  max-width: 1em;
  margin-left: auto;
  transform: ${props => (props.isOpen ? 'rotate(-90deg)' : 'rotate(0)')}
`

const NoticeElem = styled.svg`
  display: block;
  width: 1.25em;
  height: 1.25em;
  margin-right: .25em;
  fill: ${props => props.theme.highlightSecondary};
`

const TitleElem = styled.button`
  display: flex;
  padding: .5em;
  width: 100%;
  border: 0;
  margin: 0;
  font-size: 1em;
  font-weight: bold;
  font-family:'Lustria', serif;
  color: ${props => props.theme.inversePrimary};
  background: none;
  position: relative;
`

const TitleTextElem = styled.p`
  margin: 0;
`

const ContentElem = styled.div`
  padding: 1em;
  background-color: ${props => props.theme.primaryBg};
  color: ${props => props.theme.primary};
`

const DisruptionListElem = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
`
const DisruptionItemElem = styled.li`
  margin-bottom: 1em;
`

const DisruptionTitleElem = styled.span`
  display: block;
  margin-bottom: .5em;
  font-weight: bold;
`

const DisruptionDescElem = styled.span`
  display: block;
  margin-bottom: .5em;
`

const DisruptionReadMoreElem = styled.a`
  display: block;
  color: ${props => props.theme.link};
`

class Disruption extends React.Component {
  constructor() {
    super()
    this.state = {
      visible: false
    }

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick (event) {
    // Toggle
    this.setState({visible: !this.state.visible})
  }

  render() {
    let d = null
    let content = null

    if (this.props.disruptionData && this.props.disruptionData.length > 0) {
      if (this.state.visible) {
        content = <ContentElem>
          <DisruptionListElem>
          {this.props.disruptionData.map((val, index) => (
            <DisruptionItemElem key={index}>
              <DisruptionTitleElem>{val.title}</DisruptionTitleElem>
              <DisruptionDescElem>{val.description}
                {val.url ? <DisruptionReadMoreElem href={val.url} colour={val.colour}>Read more</DisruptionReadMoreElem> : null}
              </DisruptionDescElem>
            </DisruptionItemElem>
            )
          )}
          </DisruptionListElem>
        </ContentElem>
      }

      d = <ContainerElem>
      <TitleElem onClick={this.handleClick}>
        <NoticeElem viewBox="0 0 32 32">
          <path d="M15.5 3c-7.456 0-13.5 6.044-13.5 13.5s6.044 13.5 13.5 13.5 13.5-6.044 13.5-13.5-6.044-13.5-13.5-13.5zM15.5 27c-5.799 0-10.5-4.701-10.5-10.5s4.701-10.5 10.5-10.5 10.5 4.701 10.5 10.5-4.701 10.5-10.5 10.5zM15.5 10c-0.828 0-1.5 0.671-1.5 1.5v5.062c0 0.828 0.672 1.5 1.5 1.5s1.5-0.672 1.5-1.5v-5.062c0-0.829-0.672-1.5-1.5-1.5zM15.5 20c-0.828 0-1.5 0.672-1.5 1.5s0.672 1.5 1.5 1.5 1.5-0.672 1.5-1.5-0.672-1.5-1.5-1.5z"></path>
        </NoticeElem>
        <TitleTextElem>Possible disruption</TitleTextElem>
        <ChevronIconElem isOpen={this.state.visible}/>
      </TitleElem>
      {content}
    </ContainerElem>
    }

    return d
  }
}

Disruption.propTypes = {
  disruptionData: PropTypes.array
}

export default Disruption
