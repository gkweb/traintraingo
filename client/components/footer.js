import styled from 'styled-components'
import Link from 'next/link'

const FooterElem = styled.footer`
  padding: 0 1em 3em;
  margin-top: 5em;
  color: ${props => props.theme.secondary};
`

const LinkElem = styled.a`
  color: ${props => props.theme.secondary};
  &:visited {
    color: ${props => props.theme.secondary};
  }

  &:hover,
  &:focus {
    text-decoration: underline;
  }
`

const Footer = () => (
  <FooterElem>
    <p>
      Data supplied by ptv (Public Transport Victoria). Obtain your API Access{' '}
      <LinkElem href="https://www.data.vic.gov.au/data/dataset/ptv-timetable-api">
        here
      </LinkElem>
    </p>
    <LinkElem href="https://gladekettle.com.au" target="_blank">
      Created by gkweb
    </LinkElem>
  </FooterElem>
)

export default Footer
