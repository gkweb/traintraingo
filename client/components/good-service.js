import styled from 'styled-components'
import { TickIcon } from './icon'

const Container = styled.div`
  display: flex;
  flex-flow: row nowrap;
  padding: 0.5em;
`

const Icon = styled(TickIcon)`
  width: 1.25em;
  height: 1.25em;
  margin-right: 0.25em;
  fill: ${props => props.theme.highlightTertiary};
  color: ${props => props.theme.highlightTertiary};
`

const TextBlock = styled.span``

const GoodService = () => (
  <Container>
    <Icon />
    <TextBlock>Good Service</TextBlock>
  </Container>
)

export default GoodService
