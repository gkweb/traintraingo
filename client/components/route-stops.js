import { Component } from 'react'
import styled from 'styled-components'
import { H4 } from './text'
import { RefreshIcon } from './icon'
import gql from 'graphql-tag'
import { Query } from 'react-apollo'
import moment from 'moment'

const Btn = styled.button`
  display: inline-block;
  vertical-align: middle;
  padding: 0.25rem;
  border: 0;
  width: 1.5rem;
  height: 1.5rem;
  font-size: 1rem;
  text-align: center;
  background: none;
  color: inherit;
  transition: color 0.25s ease-in-out;
  cursor: pointer;
  border: 1px solid currentColor;
  border-radius: 15%;

  &:focus {
    outline: none;
  }

  &:hover,
  &:focus {
    box-shadow: 0px 0px 4px 0px rgba(0, 0, 0, 0.5);
  }
`

const RefreshBtn = ({ onClick }) => (
  <Btn onClick={onClick} aria-label="Load route departures">
    <RefreshIcon />
  </Btn>
)

const TimeItem = styled.li`
  padding: 0.5em 1em;
  position: relative;
  text-align: center;

  &:before {
    content: ' ';
    display: block;
    position: absolute;
    top: -0.5em;
    right: calc(50% - 0.25em);
    height: 0.5em;
    width: 0.5em;
    background-color: ${props => props.theme.primary};
    border-radius: 50%;
  }

  &:after {
    content: ' ';
    display: block;
    height: 0.125em;
    position: absolute;
    top: -0.25em;
    left: 0;
    right: 0;
    background-color: ${props => props.theme.primary};
  }

  &:first-child:after {
    left: 50%;
  }

  &:last-child:after {
    right: 50%;
  }
`

const Time = styled.span`
  display: block;
  font-size: 0.75rem;
`

const Station = styled.span`
  display: block;
  font-size: 0.75rem;
`

const TimeItemContainer = styled.ul`
  display: flex;
  flex-flow: row nowrap;
  padding: 1em 0 0 0;
  position: relative;
  width: 100%;
  list-style: none;
`

const Container = styled.div`
  width: 100%;
  overflow-x: auto;
  margin-bottom: 0.5em;
`

const times = [
  {
    id: '1233CF',
    time: '10:30am',
    name: 'Richmond',
  },
  {
    id: '1233FF',
    time: '11:30am',
    name: 'Richmond',
  },
  {
    id: '1233FA',
    time: '12:30pm',
    name: 'Richmond',
  },
  {
    id: '126233FA',
    time: '12:30pm',
    name: 'Richmond',
  },
  {
    id: '12sdf33FA',
    time: '12:30pm',
    name: 'Richmond',
  },
  {
    id: '12hh33FA',
    time: '12:30pm',
    name: 'Richmond',
  },
  {
    id: '12zzz3FA',
    time: '12:30pm',
    name: 'Richmond',
  },
  {
    id: '1233FD',
    time: '13:30pm',
    name: 'Richmond',
  },
]

class RouteStops extends Component {
  constructor() {
    super()
    this.state = {
      loadRouteStops: false,
    }

    this.handleLoadStops = this.handleLoadStops.bind(this)
  }

  handleLoadStops() {
    console.log('has clicked')
    this.setState({
      loadRouteStops: true,
    })
  }

  render() {
    return (
      <>
        <H4>
          Route stops <RefreshBtn onClick={this.handleLoadStops} />
        </H4>
        {this.props.run_id}
        {this.state.loadRouteStops === true ? (
          <Query
            query={STOPPING_PATTERN_QUERY}
            variables={{ run_id: this.props.runId }}
          >
            {({ loading, error, data }) => {
              let r = null
              if (loading) return <span>Loading...</span>
              if (data) {
                {
                  console.log(data)
                }
                r = (
                  <Container>
                    <TimeItemContainer>
                      {console.log(data.pattern)}
                      {data.pattern.departures.map((val, index) => {
                        const scheduled = moment(
                          val.scheduled_departure_utc
                        ).format('HH:mm')
                        const estimated = val.estimated_departure_utc
                          ? moment(val.estimated_departure_utc).format('HH:mm')
                          : null // Estimates into the future are null

                        return (
                          <TimeItem key={index}>
                            <Time>{estimated ? estimated : scheduled}</Time>
                            <Station>{val.stop_name}</Station>
                          </TimeItem>
                        )
                      })}
                    </TimeItemContainer>
                  </Container>
                )
              } else if (error) {
                {
                  console.log(error)
                }
                r = <span>{'error loading content'}</span>
              }

              return r
            }}
          </Query>
        ) : null}
      </>
    )
  }
}

const STOPPING_PATTERN_QUERY = gql`
  query($run_id: Int!) {
    pattern(run_id: $run_id) {
      departures {
        stop_id
        scheduled_departure_utc
        estimated_departure_utc
        route_id
        direction_name
        stop_name
        stop_suburb
      }
    }
  }
`

export default RouteStops
