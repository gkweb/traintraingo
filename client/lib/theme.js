import React, {Component} from 'react'

const themes = {
  light: {
    'name': 'light',
    'values': {
      'primaryBg': '#f7f7f7',
      'secondaryBg': '#ddd',
      'tertiaryBg': '#bbb',
      'inversePrimaryBg': '#f7f7f7',
      'primary': '#4c4c4c',
      'secondary': '#9a9a9a',
      'tertiary': '#f7f7f7',
      'inversePrimary': '#4c4c4c',
      'highlightPrimary': '#EEDB57',
      'highlightSecondary': '#ffB86f',
      'highlightTertiary': '',
      'logoPrimaryBg': '#5EC64C',
      'logoSecondaryBg': '#105C26',
      'logoPrimary': '#4c4c4c',
      'logoSecondary': '#fff',
      'input': '#222',
      'inputBg': '#fff',
      'inputBorder': '#ccc',
      'inputIcon': 'rgb(76, 76, 76, .75)',
      'link': '#4495d4',
      'linkBg': '#f7f7f7',
      'linkHoverBg': '#fff'
    }
  },
  dark: {
    'name': 'dark',
    'values': {
      'primaryBg': '#444',
      'secondaryBg': '#333',
      'tertiaryBg': '#222',
      'inversePrimaryBg': '#fff',
      'primary': '#fff',
      'secondary': '#ddd',
      'tertiary': '#000',
      'inversePrimary': '#000',
      'highlightPrimary': '#EEDB57',
      'highlightSecondary': '#ffB86f',
      'highlightTertiary': '',
      'logoPrimaryBg': '#5EC64C',
      'logoSecondaryBg': '#105C26',
      'logoPrimary': '#fff',
      'logoSecondary': '#fff',
      'input': '#fff',
      'inputBg': '#222',
      'inputBorder': '#000',
      'inputIcon': 'rgba(255, 255, 255, .25)',
      'link': '#fff',
      'linkBg': '#333',
      'linkHoverBg': '#222'
    }
  }
}

const ThemeManagementContext = React.createContext({
    activeTheme: themes.light.name,
    theme: themes.light // default value
  }
)

export {
  themes,
  ThemeManagementContext
}
