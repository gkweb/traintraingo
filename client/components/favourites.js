import React, {Component} from 'react'
import styled from 'styled-components'
import Link from 'next/link'
import {H3} from './text'
import {getItems} from './../lib/local-storage'
import {Bookmark} from './icon'

const FavoriteListElem = styled.ul`
  display: flex;
  flex-flow: row wrap;
  margin: 0 0 0 -.5em;
  padding: 0;
`

const FavoriteItemElem = styled.li`
  display: inline-flex;
  margin: .5em;
`

const FavoriteLinkElem = styled.a`
  display: flex;
  align-items: center;
  width: 100%;
  max-width: 30em;
  padding: 1em;
  border: 1px solid black;
  background: ${props => props.theme.primary};
  box-shadow: -.0625em .125em .5em -.125em rgba(0,0,0,.7);
  color: ${props => props.theme.tertiary};
  text-decoration: none;
  transition: transform .125s ease;

  &:hover,
  &:focus {
    transform: scale(1.05);
  }

  span:last-child {
    margin-left: .5em;
    color: ${props => props.theme.highlightPrimary};
  }
`

const FavouritesContainerElem = styled.section`
  padding: 1em 1em 4em;
  background-color: #fff;

  @media screen and (min-width: 50em) {
    padding: 1em 1em 4em;
    margin-top: 3em;
  }
`

class Favourites extends React.Component {
  constructor() {
    super()
    this.state = {
      favourites: {}
    }
  }

  componentDidMount() {
    const favourites = getItems()
    if (favourites) this.setState({favourites: favourites})
  }

  render() {
    let r

    if (this.state.favourites && Object.keys(this.state.favourites).length > 0) {
      r = <FavouritesContainerElem>
      <H3>Favourites</H3>
      <FavoriteListElem>
        {Object.keys(this.state.favourites).map((val, ind) => (
          <FavoriteItemElem key={ind}>
            <Link href={`/departures/${val}`} as={`/departures/${val}`} passHref>
              <FavoriteLinkElem>
                <span>{this.state.favourites[val]}</span>
                <span><Bookmark/></span>
              </FavoriteLinkElem>
            </Link>
        </FavoriteItemElem>
        ))}
      </FavoriteListElem>
      </FavouritesContainerElem>
    } else {
      r = null
    }

    return r
  }
}

export default Favourites
