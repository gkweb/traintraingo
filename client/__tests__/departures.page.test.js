import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import { themes } from './../lib/theme'
import Departures, { STOP_QUERY } from '../pages/departures'
import { STOPPING_PATTERN_QUERY } from '../components/route-stops'
import 'jest-styled-components'
import { MockedProvider } from 'react-apollo/test-utils'
import renderer from 'react-test-renderer'
import waitForExpect from 'wait-for-expect'

// Mutate deps
import * as localStorage from './../lib/fav-local-storage'
jest.mock('./../lib/fav-local-storage', () => ({
  updateItem: jest.fn(),
  removeItem: jest.fn(),
  getItem: jest.fn(),
}))

// Helps with Link.prefetch mocking
jest.mock('next/link', () => {
  return ({ children }) => {
    return children
  }
})

const MockStation = {
  stop: {
    stop_id: '1108',
    stop_name: 'Kensington',
    departures: [
      {
        stop_id: '1108',
        run_id: '12345',
        platform_number: 1,
        route_id: 3,
        direction_id: 1,
        estimated_departure_utc: '2019-04-23T09:05:00Z',
        scheduled_departure_utc: '2019-04-23T09:01:00Z',
        direction_name: 'City (Flinders Street)',
        disruptions: [
          {
            sid: 'FcKK908',
            title:
              'Craigieburn, Sunbury, Upfield, Werribee and Williamstown: No City Loop services on Saturday 4 May and Sunday 5 May 2019',
            description:
              'Craigieburn, Sunbury, Upfield, Werribee and Williamstown line trains will run direct to and from FlindersÂ Street Station, not via the City Loop on Saturday 4 May and Sunday 5 May 2019.',
            url:
              'http://ptv.vic.gov.au/live-travel-updates/article/craigieburn-sunbury-upfield-werribee-and-williamstown-no-city-loop-services-on-saturday-4-may-and-sunday-5-may-2019',
            display_status: false,
            colour: '#ffd500',
          },
        ],
        run: {
          run_id: 12345,
          express_stop_count: 1,
          destination_name: 'Laverton',
        },
      },
      {
        stop_id: '1108',
        platform_number: 2,
        run_id: '12346',
        route_id: 3,
        direction_id: 2,
        estimated_departure_utc: '2019-04-23T09:20:00Z',
        scheduled_departure_utc: '2019-04-23T09:20:00Z',
        direction_name: 'Craigieburn',
        disruptions: [],
        run: {
          run_id: 12346,
          express_stop_count: 0,
          destination_name: 'Craigieburn',
        },
      },
    ],
  },
}

// Stops Mock
const mocks = [
  {
    request: {
      query: STOP_QUERY,
      variables: {
        stop_id: 1108,
      },
    },
    result: {
      data: MockStation,
    },
  },
]

test('Displays Loading text', () => {
  const c = mount(
    <ThemeProvider theme={themes.light.values}>
      <MockedProvider mocks={[]} addTypename={false} removeTypename>
        <Departures stop_id={1108} />
      </MockedProvider>
    </ThemeProvider>
  )

  expect(c.text()).toContain('Loading...')
})

test('Displays station name as text', async () => {
  const c = renderer.create(
    <ThemeProvider theme={themes.light.values}>
      <MockedProvider
        mocks={mocks}
        variables={{ stop_id: 1108 }}
        addTypename={false}
        removeTypename
      >
        <Departures stop_id={1108} />
      </MockedProvider>
    </ThemeProvider>
  )

  await waitForExpect(() => {
    expect(JSON.stringify(c.toJSON())).toMatch('Kensington')
  })
})

test('Departures Snapshot', () => {
  const c = shallow(<Departures stop_id={1108} />, {
    theme: themes.light.values,
  })
  expect(c).toMatchSnapshot()
})
