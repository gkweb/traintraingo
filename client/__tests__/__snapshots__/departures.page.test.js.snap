// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Departures Snapshot 1`] = `
<Query
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": undefined,
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "stop_id",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "stop_id",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "stop",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "stop_id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "stop_name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "departures",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "stop_id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "platform_number",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "route_id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "direction_id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "estimated_departure_utc",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "scheduled_departure_utc",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "direction_name",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "disruptions",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "title",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "description",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "url",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "display_status",
                                  },
                                  "selectionSet": undefined,
                                },
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "colour",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "Int",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "stop_id",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 415,
        "start": 0,
      },
    }
  }
  variables={
    Object {
      "stop_id": 1108,
    }
  }
>
  <Component />
</Query>
`;
